tests/js/app/modules/filter/testType.js

```diff
@@ -0,0 +1,51 @@
+// Copyright 2018 Endless Mobile, Inc.
+
+const Eknc = imports.gi.EosKnowledgeContent;
+
+const Type = imports.app.modules.filter.type;
+const MockFactory = imports.tests.mockFactory;
+
+describe('Filter.Type', function () {
+    const MODELS = ['text/html', 	'video/jpeg'].map(type =>
+        Eknc.ContentObjectModel.new_from_props({
+            content_type: type,
+        }));
+    let filter;
+
+    describe('normal mode', function () {
+        beforeEach(function () {
+            [filter] = MockFactory.setup_tree({
+                type: Type.Type,
+                properties: {
+                    type: 'video/jpeg',
+                },
+            });
+        });
+
+        it('is the default', function () {
+            expect(filter.invert).toBeFalsy();
+        });
+
+        it('filters out a non-video models', function () {
+            expect(filter.include(MODELS[0])).toBeFalsy();
+            expect(filter.include(MODELS[1])).toBeTruthy();
+        });
+    });
+
+    describe('invert mode', function () {
+        beforeEach(function () {
+            [filter] = MockFactory.setup_tree({
+                type: Type.Type,
+                properties: {
+                    type: 'video/jpeg',
+                    invert: true,
+                },
+            });
+        });
+
+        it('filters out a non-video models (behavior unaltered)', function () {
+            expect(filter.include(MODELS[0])).toBeFalsy();
+            expect(filter.include(MODELS[1])).toBeTruthy();
+        });
```

Comment:
Perhaps it'd be better to have an `xit` test here to test the expected behaviour once the `invert` property works correctly.

Reply:
Well, if we end-up adding `excluded_content_type` (like we mentioned above), I will change it anyways.